Domain/Shipping/Cargo/Itinerary.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Domain.Shipping.Location;
+using Domain.Shipping.Voyage;
+
+namespace Domain.Shipping.Cargo
+{
+    public class Itinerary 
+    {
+        public IReadOnlyCollection<Leg> Legs { get; }
+
+        public Itinerary(IList<Leg> legs)
```

Comment:
```suggestion
        public Itinerary(IReadOnlyCollection<Leg> legs)
```

Reply:
Maybe I'm not sure. What IReadOnlyCollections allow? You can enlight me on that as I'm not so familiar with lists.

Reply:
Or `IReadOnlyList<Leg>` perhaps. Just to be explicit that we're injecting a list and we have no plans to write to it (read only).

Reply:
IReadOnlyList<T> accepts IList<T> from the client?